src/codec.rs

```diff
@@ -184,7 +184,9 @@ impl TryFrom<sys::uhid_event> for OutputEvent {
         if let Some(event_type) = to_uhid_event_type(event.type_) {
             match event_type {
                 sys::uhid_event_type_UHID_START => Ok(unsafe {
-                    let payload = &event.u.start;
+                    // Deal with unaligned packed struct access
+                    let payload_rw = std::ptr::addr_of!(event.u.start);
+                    let payload = payload_rw.read_unaligned();
```

Comment:
FWIW you can just do `let payload = event.u.start;`. Or better, just look at `event.u.start.dev_flags` to avoid an unnecessary big copy.
 Direct field accesses to packed fields are fine and the compiler knows how to handle their lack of alignment. Only taking a reference via `&` is a problem.

Reply:
Thanks for the tip!